src/webui/abstractwebapplication.cpp

```diff
@@ -354,6 +354,7 @@ bool AbstractWebApplication::sessionStart()
         sessions_[session_->id] = session_;
 
         QNetworkCookie cookie(C_SID, session_->id.toUtf8());
+        cookie.setHttpOnly(true);
```

Comment:
Tested, it's working fine with https.

I should have mention this earlier, from: https://www.owasp.org/index.php/HttpOnly#What_is_HttpOnly.3F
>If the HttpOnly flag (optional) is included in the HTTP response header, the cookie cannot be accessed through client side script 

---

Also what you said should be secure cookie
https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies